### PR TITLE
feat: add 0.5 release blog

### PR DESCRIPTION
**Pull request by ahabhgk**

### CODE REVIEW

File: docs/en/blog/announcing-0.5.mdx [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+## Major Feature Updates
+
+### Module Federation added to Rspack
+
+Checkout [this blog](/blog/module-federation-added-to-rspack) for more details.
+
+## Breaking Changes
+
+### optimization.chunkIds is deterministic in production mode by default
```
Comment (h-a-n-a) [inline docs/en/blog/announcing-0.5.mdx:17]:
```suggestion
### optimization.chunkIds is 'deterministic' in production mode by default
```

File: docs/en/blog/announcing-0.5.mdx [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+## Migration Guide
+
+v0.5.0 removed lots of deprecated features, except that, v0.5.0 introduced four breaking changes, and you only need to notice two of them if you are developing applications using Rspack. So v0.5.0 is easy to migrate if you already migrate to v0.4+ with no deprecate warnings, if you haven't, checkout the [v0.4.0 migration guide](https://www.rspack.dev/blog/announcing-0.4#migration-guide).
+
+### Add resolve.extensions
+
+This is a breaking change that is most likely to affect you.
+
+After you upgrade `@rspack/core` to v0.5.0, if you build failed with error: `Failed to resolve ./src/xxx.tsx`, or `Failed to resolve ./src/xxx.ts`, or `Failed to resolve ./src/xxx.jsx`, you need to add `resolve.extensions = ['...', '.tsx', '.ts', '.jsx']` in your configuration.
```
Comment (h-a-n-a) [inline docs/en/blog/announcing-0.5.mdx:193]:
```suggestion
After you upgrade `@rspack/core` to v0.5.0, if you build failed with error: `Can't resolve './src/xxx.tsx'`, or `Can't resolve './src/xxx.ts'`, or `Can't resolve './src/xxx.jsx'`, you need to add `resolve.extensions = ['...', '.tsx', '.ts', '.jsx']` in your configuration.
```

File: docs/zh/blog/announcing-0.5.mdx [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+
+## 主要功能更新
+
+### Rspack 支持模块联邦
+
+请查看[此博客](/blog/module-federation-added-to-rspack)以获取更多详细信息。
+
+## 破坏性更新
+
+### 调整 optimization.chunkIds 生产环境默认值为 deterministic
```
Comment (h-a-n-a) [inline docs/zh/blog/announcing-0.5.mdx:17]:
```suggestion
### 调整 optimization.chunkIds 生产环境默认值为 "deterministic"
```

File: docs/zh/blog/announcing-0.5.mdx [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+
+## 迁移指南
+
+v0.5.0 移除了许多已经弃用的功能，除此之外，v0.5.0 引入了四个破坏性更新，如果您正在使用 Rspack 开发应用，那么您只需要关注其中两个。因此，如果您已经迁移到 v0.4+ 并且没有弃用警告，那么 v0.5.0 将很容易迁移，如果您还没有，请查看 [v0.4.0 迁移指南](https://www.rspack.dev/blog/announcing-0.4#migration-guide).
+
+### 添加 resolve.extensions
+
+这是一个很可能会影响到您的破坏性更新。
+
+在将 `@rspack/core` 升级到 v0.5.0 后，如果构建失败并出现错误：`Failed to resolve ./src/xxx.tsx`，或者 `Failed to resolve ./src/xxx.ts`，或者 `Failed to resolve ./src/xxx.jsx`，则需要在配置中添加 `resolve.extensions = ['...', '.tsx', '.ts', '.jsx']`。
```
Comment (h-a-n-a) [inline docs/zh/blog/announcing-0.5.mdx:193]:
```suggestion
在将 `@rspack/core` 升级到 v0.5.0 后，如果构建失败并出现错误：`Can't resolve './src/xxx.tsx'`，或者 `Can't resolve './src/xxx.ts'`，或者 `Can't resolve './src/xxx.jsx'`，则需要在配置中添加 `resolve.extensions = ['...', '.tsx', '.ts', '.jsx']`。
```

File: docs/zh/blog/announcing-0.5.mdx (hunk 3 above)
Comment (ahabhgk) [reply docs/zh/blog/announcing-0.5.mdx:17]: 别加引号了，Rspress 标题加引号会带上个 \

**Review by hardfist — APPROVED** [review]

**Review by hardfist — APPROVED** [review]

**Pull request merged**